src/CalcManager/CalculatorHistory.cpp

```diff
@@ -7,25 +7,46 @@
 using namespace std;
 using namespace CalculationManager;
 
+namespace
+{
+    static wstring GetGeneratedExpression(const vector<pair<wstring, int>>& tokens)
+    {
+        wstring expression;
+        const size_t nTokens = tokens.size();
+
+        for (size_t i = 0; i < nTokens; ++i)
```

Comment:
NIT: The original code was this way, but we can probably simplify it using `for (auto const& s : tokens)` instead

Reply:
I didn't update this to a range based for due to the check index check happening inside the loop so that it inserts a space after each token except for the last token.

Reply:
what do you think about this code?
```
bool isFirst = true;
for (auto const& token : tokens)
{
  if(isFirst)
  {
    isFirst = false;
    expression += L' ';
  }  
  expression.append(token.first);
}
```

Reply:
That works.  This is where I wish the C++ standard library had some sort of Join function.

Reply:
Yes, I wrote the code from my phone, you are right, I was thinking about:
```
bool isFirst = true;
for (auto const& token : tokens)
{
  expression.append(token.first);
  if(isFirst)
  {
    isFirst = false;
    expression += L' ';
  }  
}
```

but your version works too! 

Based on old screenshots, it was also the case of the current operation part (above the result). We can keep it this way for the moment but I will open a new issue so we can chat about it.

![image](https://user-images.githubusercontent.com/1226538/67882551-a9917f00-faff-11e9-9553-7c3406106c77.png)